### PR TITLE
Subtopic Content Update [DO NOT MERGE] [DO NOT REVIEW]

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation

This PR is just a reference to seek some clarification on future work items.

Some major issues in this PR:
 - In this PR, you will notice that I have hard-coded the `topicId` in `RevisionCardViewModel` because that id is needed to fetch the image from oppia server.
- If you notice the `fractions_subtopic_0.json` file, you will see that the format of json content has changed drastically.
- Along with this, we need to update the `Topic-Revision` tab too to contain images from server, which means that will also, need topic-id which is available on oppia server. Same applies for exploration, topic, subtopic, skills, stories, etc.

Now, I have two suggestions to achieve all things which are mentioned above correctly:
1. We keep `First Topic`, `Second Topic`, `Fractions` and `Ratios` for testing purpose in which we don't need to worry about images from the oppia-server.
2. We introduce one new topic (e.g `Place Values`) and make all the necessary changes to `model` -> `domain` -> `app` layer such that it supports all new json parsing and also supports images from oppia-server.
3. Now, we update the internal content of the `Place Values` in json files such that it now looks like a test project but contains images which are available on oppia server.

Alternate Approach:
For `First Topic` we update the topic id from `test_topic_id_0` to `iX9kYCjnouWN` (Place Values on oppia-server) so that it
supports all images which need `topic_id` (like subtopic/revisionCard).

The result of this current PR is:

## Revision Card Screenshot
Screenshot_1594757955

## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [ ] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [ ] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [ ] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: domain/src/main/assets/fractions_subtopic_0.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "subtopic_title": "What is Fraction?",
+  "next_subtopic_dict": {
```
Comment (rt4914) [inline domain/src/main/assets/fractions_subtopic_0.json:3]: @BenHenning This is the major change in this new json parsing structure, but I amnot sure how much important this is for us.

Comment (BenHenning) [reply domain/src/main/assets/fractions_subtopic_0.json:3]: I'm not actually sure what this is for--are users expected to cycle through multiple revision cards at a time?

**Comment by BenHenning** [comment]
Thanks @rt4914--some questions & thoughts:
1. We definitely shouldn't be checking in any of the new topics structures in Oppia Android, and should actually be removing the fractions & ratios ones soon. I'm guessing this is PR is just meant to be an example, though?
2. Why do the structures need to change for images to work? I'd expect them just to work out of the box. Am I missing something?
3. For loading the lessons, I think we should try to make this a bit more generic like we discussed last week. In the interim, we could have something like a topics.json file that actually lists the topic IDs available to load and use that to populate the topic list. Otherwise, the app should automatically load all JSON files it has access to based on the available topic IDs. This effectively makes all elements of a topic plug-and-play. If the topic data is included & its ID is in that list, it will automatically work in the app. This will provide a nice stepping stone for full download support, and it will allow the app to operate both in a 'developer' and 'alpha' mode based on what JSON files we include.

File: domain/src/main/assets/fractions_subtopic_0.json (hunk 1 above)
Comment (rt4914) [reply domain/src/main/assets/fractions_subtopic_0.json:3]: After checking the web, https://www.oppia.org/subtopic/Multiplication/1, this content is used to show next card.

Screenshot 2020-07-15 at 2 20 56 AM

**Comment by rt4914** [comment]
> Thanks @rt4914--some questions & thoughts:
> 
> 1. We definitely shouldn't be checking in any of the new topics structures in Oppia Android, and should actually be removing the fractions & ratios ones soon. I'm guessing this is PR is just meant to be an example, though?
> 2. Why do the structures need to change for images to work? I'd expect them just to work out of the box. Am I missing something?
> 3. For loading the lessons, I think we should try to make this a bit more generic like we discussed last week. In the interim, we could have something like a topics.json file that actually lists the topic IDs available to load and use that to populate the topic list. Otherwise, the app should automatically load all JSON files it has access to based on the available topic IDs. This effectively makes all elements of a topic plug-and-play. If the topic data is included & its ID is in that list, it will automatically work in the app. This will provide a nice stepping stone for full download support, and it will allow the app to operate both in a 'developer' and 'alpha' mode based on what JSON files we include.

1. Okay, so in that, do we plan to remove fractions / ratios before alpha release? Because if we do that, there will updates needed in test cases which uses these topics for testing.

2. I think you are correct, images should work directly because of `thumbnail_url` along with `LessonThumbnail`. But 
lets take one example, we have a `initialise.json` file in `domain/assets` which lists down all the file-names and their corresponding id along with this the list of all topic ids. So that whenever a `topic_id` is parsed and we got to child content like `subtopic_id`, we know in which file we can see for the content of that `subtopic_id`. Now this file will act as the entry point and rest all content will be fetched dynamically. So in this case we won't need any topic/subtopic/skill/story/exploration id which we have hard coded currently in `TopicController` & `TopicListController`.  Now, in this case the images will work correctly but I think we won't be able to support LessonThumbnails the way we are doing right now like `createSubtopicThumbnail`, `createTopicThumbnail`, etc because the code does not contain any ids and everything is in `initialise.json`, we can show placeholder thumbnails if the image-url fails to work because of internet.

3. I think my 2nd point does explain this in detail, correct me if this is wrong.

**Comment by rt4914** [comment]
> > Thanks @rt4914--some questions & thoughts:
> > 
> > 1. We definitely shouldn't be checking in any of the new topics structures in Oppia Android, and should actually be removing the fractions & ratios ones soon. I'm guessing this is PR is just meant to be an example, though?
> > 2. Why do the structures need to change for images to work? I'd expect them just to work out of the box. Am I missing something?
> > 3. For loading the lessons, I think we should try to make this a bit more generic like we discussed last week. In the interim, we could have something like a topics.json file that actually lists the topic IDs available to load and use that to populate the topic list. Otherwise, the app should automatically load all JSON files it has access to based on the available topic IDs. This effectively makes all elements of a topic plug-and-play. If the topic data is included & its ID is in that list, it will automatically work in the app. This will provide a nice stepping stone for full download support, and it will allow the app to operate both in a 'developer' and 'alpha' mode based on what JSON files we include.
> 
> 1. Okay, so in that, do we plan to remove fractions / ratios before alpha release? Because if we do that, there will updates needed in test cases which uses these topics for testing.
> 2. I think you are correct, images should work directly because of `thumbnail_url` along with `LessonThumbnail`. But
>    lets take one example, we have a `initialise.json` file in `domain/assets` which lists down all the file-names and their corresponding id along with this the list of all topic ids. So that whenever a `topic_id` is parsed and we got to child content like `subtopic_id`, we know in which file we can see for the content of that `subtopic_id`. Now this file will act as the entry point and rest all content will be fetched dynamically. So in this case we won't need any topic/subtopic/skill/story/exploration id which we have hard coded currently in `TopicController` & `TopicListController`.  Now, in this case the images will work correctly but I think we won't be able to support LessonThumbnails the way we are doing right now like `createSubtopicThumbnail`, `createTopicThumbnail`, etc because the code does not contain any ids and everything is in `initialise.json`, we can show placeholder thumbnails if the image-url fails to work because of internet.
> 3. I think my 2nd point does explain this in detail, correct me if this is wrong.

For 2nd point, another approach can the id-name which was suggested by you.
There will be one `topics.json` file which contains list of topic ids and all other json files will contain the id-name that they correspond to, so that we don't need to have id-to-file mapping.

**Comment by BenHenning** [comment]
Your approach sounds good (either approach for what we store in the file--the goal here is to not have any hardcoded IDs or mappings in TopicController/TopicListController anymore). As discussed, we will still have fractions & ratios in the test app for now (and will remove these once we have a longer term developer strategy built), but I think what that looks like is the dev-only manifest JSON file will contain all of the test IDs/references. By replacing this & the corresponding assets with the production versions, we'll be removing the test assets & their availability from the build.

Does that help clarify?

**Comment by rt4914** [comment]
> Your approach sounds good (either approach for what we store in the file--the goal here is to not have any hardcoded IDs or mappings in TopicController/TopicListController anymore). As discussed, we will still have fractions & ratios in the test app for now (and will remove these once we have a longer term developer strategy built), but I think what that looks like is the dev-only manifest JSON file will contain all of the test IDs/references. By replacing this & the corresponding assets with the production versions, we'll be removing the test assets & their availability from the build.
> 
> Does that help clarify?

Everything is clear. Thanks. Will be closing this PR as we discussed we don't need to check the image content for revision card separately. Thanks.

**Pull request closed without merging**